medmodels/medrecord/_overview.py

```diff
@@ -1,109 +1,134 @@
-from typing import Dict, List, Optional, Union
+from typing import Dict, List, Literal, Optional, Union
 
 import polars as pl
 
 from medmodels.medrecord.schema import AttributesSchema, AttributeType
-from medmodels.medrecord.types import Attributes, EdgeIndex, NodeIndex
+from medmodels.medrecord.types import (
+    AttributeInfo,
+    Attributes,
+    EdgeIndex,
+    Group,
+    MedRecordAttribute,
+    NodeIndex,
+)
 
 
 def extract_attribute_summary(
-    attribute_dict: Union[Dict[EdgeIndex, Attributes], Dict[NodeIndex, Attributes]],
+    attribute_dictionary: Union[
+        Dict[EdgeIndex, Attributes], Dict[NodeIndex, Attributes]
+    ],
     schema: Optional[AttributesSchema] = None,
     decimal: int = 2,
-) -> pl.DataFrame:
+) -> Dict[MedRecordAttribute, List[str]]:
```

Comment:
The `Listr[str]` is not machine readable. The return type should at least be `Dict[MedRecordAttribute, Dict[Literal["min", "max", "mean", etc.], MedRecordValue]`